src/node/fetch/fetch.node.ts

```diff
@@ -239,4 +234,22 @@ export class NodeFetch implements IFetch, Interceptable, AuthCache {
 
     return false;
   }
+
+  private prepareHeadersInit(data: FetchParameters['headers'] | undefined): HeadersInit {
+    if (data === undefined) {
+      return [];
+    }
+
+    const headers: [string, string][] = [];
+
+    Object.entries(data).forEach(([key, value]) => {
+      if (Array.isArray(value)) {
+        value.forEach((val) => headers.push([key, val]));
+      } else {
+        headers.push([key, value]);
+      }
+    });
```

Comment:
thought: Would using `flatMap` make this more readable?

```suggestion
    headers = Object.entries(data).flatMap(([key, value]) => {
      if (Array.isArray(value)) {
        return value.map((nestedVal) => [key, nestedVal])
      }
      return [[key, value]]
    }
```

But it's probably less performant (but so is `forEach`).

Reply:
Thank you, but I intentionally implemented this way.

Reply:
I think at that point I would find a classic `for .. of` loop more readable, but I think this code is short enough to be fine either way.